### ISSUE TITLE
Multi-label custom training

### ISSUE THREAD

**Issue by segalinc** [ultralytics/yolov5#1811]
## 
I am interested in training yolov5 for multi-label on my own dataset. Let's say I have 3 classes where 2 range (0,1) and the third (0,7)
how can I modify the yaml files to do this?
Do I also need to edit anything in the model architecture and loss then? I think the loss should be fine as it's BCELoss

**Comment by glenn-jocher** [comment]
@segalinc multi-label inference is enabled by default. To train with multiple labels, simply add additional rows to your labels.txt files with any additional classes for the same box.

**Comment by segalinc** [comment]
Thank you I will try it !

**Comment by segalinc** [comment]
I was just taking a look at other post about this and my request might be different as my labels are actually the scores not the categorical labels of the classes, so they are continuous. Only the third class is a an actual classification over categorical classes the other 2 are more a regression

**Comment by glenn-jocher** [comment]
@segalinc you may want to provide links/documentation to what you're referring to.

**Comment by segalinc** [comment]
I'd like to reproduce this paper https://www.sciencedirect.com/science/article/pii/S1877050920305895
Basically they use yolo (v2) to solve a multitask problem of predicting both categorical emotion and continuous value for valence and arousal given an image. The categorical part is already there, the continuous one I am not sure how to prepare the data here and I guess I will need to edit a bit the code

**Comment by glenn-jocher** [comment]
@segalinc oh interesting. Well you can think of YOLOv5 mainly as a feature extractor with a specific final output layer that's pliable to a variety of tasks. There's already an existing implementation of predicting continuous values in the bounding box regression that YOLOv5 does. The criteria there is based on IoU, but it can also be based off of MSE loss, as implemented in the original darknet version of YOLOv3.

Anything you want to do though will require a correspondingly labelled dataset naturally, since this is supervised training.

**Comment by alicera** [comment]
If I have 2 labels,  how many object can be detected for max?

**Comment by glenn-jocher** [comment]
@alicera I don't understand, can you rephrase?

Theoretically your model can detect a number of objects equal to the number of anchors in your image (i.e. several thousand).

**Comment by alicera** [comment]
For 640x640 image and design 2 label.
It will have the objects of (80x80+40x40+20x20) * 3 anchors * 2 class can be detected, right?

**Comment by glenn-jocher** [comment]
Sure.

**Comment by segalinc** [comment]
just a followup feedback: in order to perform regression there are lot of things to change in the code as everything is set up for classification (class weights init, image_weights, plots, performances, loss, data loaders)

**Comment by utkutpcgl** [comment]
Beware the labels are one_hot (which should be actually n_hot where n are is the number of predicted or ground truth labels). You can check: https://github.com/ultralytics/yolov5/issues/1435#issuecomment-867521827
